src/boot/efi/stub.c

```diff
@@ -428,9 +429,10 @@ static EFI_STATUS load_addons(
                 if (err != EFI_SUCCESS)
                         return log_error_status(err, "Failed to find protocol in %ls: %m", items[i]);
 
-                err = pe_memory_locate_sections(loaded_addon->ImageBase, unified_sections, addrs, szs);
+                err = pe_memory_locate_sections(loaded_addon->ImageBase, &sections, &n_sections);
                 if (err != EFI_SUCCESS ||
-                    (szs[UNIFIED_SECTION_CMDLINE] == 0 && szs[UNIFIED_SECTION_DTB] == 0)) {
+                    (!pe_bsearch_section(sections, n_sections, unified_sections[UNIFIED_SECTION_CMDLINE]) &&
```

Comment:
maybe add a helper around `pe_find_section()` called `pe_find_unified_section()` and takes a UnifiedSection instead of a const char* for the section name? would mae this more readable here